src/iop/atrous.c

```diff
@@ -249,60 +253,49 @@ static void process_wavelets(struct dt_iop_module_t *self, struct dt_dev_pixelpi
     return;
   }
 
-  float *detail[MAX_NUM_SCALES] = { NULL };
-  float *tmp = NULL;
-  float *buf2 = NULL;
-  float *buf1 = NULL;
+  float *const restrict out = (float*)o;
+  float *restrict detail = NULL;
+  float *restrict tmp = NULL;
+  float *restrict tmp2 = NULL;
 
-  tmp = (float *)dt_alloc_align_float((size_t)4 * width * height);
-  if(tmp == NULL)
+  if (!dt_iop_alloc_image_buffers(self, NULL, roi_in, roi_out,
```

Comment:
this call seems out of sync with the imagebuf definition:

```
gboolean dt_iop_alloc_image_buffers(struct dt_iop_module_t *const module,
                                    const struct dt_iop_roi_t *const roi_in,
                                    const struct dt_iop_roi_t *const roi_out, ...);
```

The NULL before roi_in is suspicious!

Reply:
Oops, missed that one because it didn't generate a merge conflict.